### PR TITLE
myriad/fix_term_signal

### PR DESCRIPTION
**Pull request by eyadof**
Fix TERM signal context by using curry
Reduce log verbosity

### CODE REVIEW

File: lib/Myriad.pm [hunk 1]
```diff
@@ -439,18 +440,18 @@ Applies signal handlers for TERM and QUIT, then starts the loop.
...
         $log->infof('INT received, exit');
         $self->shutdown->await
-    });
-    $loop->attach_signal(QUIT => async method {
+    }));
+    $loop->attach_signal(QUIT => $self->$curry::weak(method {
         $log->infof('QUIT received, exit');
         $self->shutdown->await
-    });
+    }));
```
Comment (zakame) [inline lib/Myriad.pm:454]: Maybe rephrase-able as

```perl
map { 
    $loop->attach_signal($_ => $self->$curry::weak(method {
        $log->infof("%s received, exit", $_);
        $self->shutdown->await;
    }))
} qw(TERM INT QUIT);
```

**Review by nael-binary — APPROVED** [review]

File: lib/Myriad.pm (hunk 1 above)
Comment (ea-binary) [reply lib/Myriad.pm]: with a bit of tweak since `$_` is not accessible inside the `method` block.

**Review by zakame — APPROVED** [review]
